x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,353 @@
+  const { logger } = opts;
+  const loggerTag = claimAvailableTasksMget.name;
+  const logMeta = { tags: [loggerTag] };
+  const initialCapacity = getCapacity();
+
+  const removedTypes = new Set(unusedTypes); // REMOVED_TYPES
+  const excludedTypes = new Set(excludedTaskTypes); // excluded via config
+  const { docs, versionMap } = await searchAvailableTasks({
+    definitions,
+    taskTypes: new Set(definitions.getAllTypes()),
+    excludedTypes,
+    removedTypes,
+    taskStore,
+    events$,
+    claimOwnershipUntil,
+    size: initialCapacity * SIZE_MULTIPLIER_FOR_TASK_FETCH,
+    taskMaxAttempts,
+  });
+
+  if (docs.length === 0) return emptyClaimOwnershipResult();
+
+  const currentTasks = new Set<ConcreteTaskInstance>();
+  const staleTasks = new Set<ConcreteTaskInstance>();
+  const missingTasks = new Set<ConcreteTaskInstance>();
+  const removedTasks = new Set<ConcreteTaskInstance>();
+
+  const docLatestVersions = await taskStore.getDocVersions(docs.map((doc) => `task:${doc.id}`));
+
+  for (const searchDoc of docs) {
+    if (removedTypes.has(searchDoc.taskType)) {
+      removedTasks.add(searchDoc);
+      continue;
+    }
+
+    const searchVersion = versionMap.get(searchDoc.id);
+    const latestVersion = docLatestVersions.get(`task:${searchDoc.id}`);
+    if (!searchVersion || !latestVersion) {
+      missingTasks.add(searchDoc);
+      continue;
+    }
+
+    if (
+      searchVersion.seqNo === latestVersion.seqNo &&
+      searchVersion.primaryTerm === latestVersion.primaryTerm
+    ) {
+      currentTasks.add(searchDoc);
+      continue;
+    } else {
+      staleTasks.add(searchDoc);
+      continue;
+    }
+  }
+
+  for (const doc of missingTasks) {
+    logger.info(`Task ${doc.id} is missing from the task store or something`, logMeta);
+  }
+
+  for (const doc of staleTasks) {
+    logger.info(`Task ${doc.id} is stale`, logMeta);
+  }
+
+  const candidateTasks = applyLimitedConcurrency(currentTasks, batches);
+  const taskUpdates: ConcreteTaskInstance[] = Array.from(candidateTasks)
+    .slice(0, initialCapacity)
+    .map((task) => {
+      if (task.retryAt != null && new Date(task.retryAt).getTime() < Date.now()) {
+        task.scheduledAt = task.retryAt;
+      } else {
+        task.scheduledAt = task.runAt;
+      }
+      task.retryAt = claimOwnershipUntil;
+      task.ownerId = taskStore.taskManagerId;
+      task.status = TaskStatus.Claiming;
+
+      return task;
+    });
+
+  const finalResults: ConcreteTaskInstance[] = [];
+  let conflicts = staleTasks.size;
+  try {
+    const updateResults = await taskStore.bulkUpdate(taskUpdates, { validate: false });
+    for (const updateResult of updateResults) {
+      if (isOk(updateResult)) {
+        finalResults.push(updateResult.value);
+      } else {
+        conflicts++;
+        const { id, type, error } = updateResult.error;
+        logger.warn(`Error updating task ${id}:${type} during claim: ${error.message}`, logMeta);
+      }
```

Comment:
We should check for `SavedObjectsErrorHelpers.isConflictError` before marking it as a conflict. For non-conflict errors, the update by query use to throw but I'm not sure what happens with bulk, maybe it never throws but each doc has something?

Reply:
During a 7hr run today, I saw 1M stale entries, and 27K errors in the items in the bulk updates.  All the item errors in bulk were conflicts.

The result of having a conflict is that - currently - we just ignore it.  I think longer term we'll do one more bulk update (prolly just one) with some of the tasks not selected for claiming, to fill in those "holes".  I don't think we'd consider conflict vs some other error to be any different for that purpose.

I'll add some code to produce a different error for these, but otherwise I think the behavior will be the same - basically just "skipping" processing of that task ...

Reply:
I added the SO conflict check in 0c5c83c94059668eaa47d030a8e1f8fec6a83c07 (forgot to mention it in the commit comment!).  Kinda weird how I had to check twice, because of the somewhat odd typing of the SO errors ...

Reply:
Will the task claim failure metrics capture the non conflict errors? That the main difference I can think of when it comes to conflict vs non-conflict errors, we may need to act on the non-conflict ones (or see what they could be!).

Reply:
Ya, will need to look into that - I don't think we have any metrics there for the default claimer.  We also don't have any place to put the "stale" entries.  I was combining "stale" and "conflicts" together, but that's not really useful for me now.  Maybe that's ok for production?  They are kinda the same thing ...